src/ynEIGEN/EigenStrategyManager.sol

```diff
@@ -0,0 +1,294 @@
+    mapping(IERC20 => IStrategy) public strategies;
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  INITIALIZATION  ----------------------------------
+    //--------------------------------------------------------------------------------------
+
+    struct Init {
+        IERC20[] assets;
+        IStrategy[] strategies;
+        IynEigen ynEigen;
+        IStrategyManager strategyManager;
+        IDelegationManager delegationManager;
+        ITokenStakingNodesManager tokenStakingNodesManager;
+        address admin;
+        address strategyController;
+        address unpauser;
+        address pauser;
+    }
+
+    function initialize(Init calldata init)
+        public
+        notZeroAddress(address(init.strategyManager))
+        notZeroAddress(address(init.admin))
+        notZeroAddress(address(init.strategyController))
+        initializer {
+        __AccessControl_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.pauser);
+        _grantRole(UNPAUSER_ROLE, init.unpauser);
+        _grantRole(STRATEGY_CONTROLLER_ROLE, init.strategyController);
+
+        for (uint256 i = 0; i < init.assets.length; i++) {
+            if (address(init.assets[i]) == address(0) || address(init.strategies[i]) == address(0)) {
+                revert ZeroAddress();
+            }
+            strategies[init.assets[i]] = init.strategies[i];
+        }
+
+        ynEigen = init.ynEigen;
+
+        strategyManager = init.strategyManager;
+        delegationManager = init.delegationManager;
+        tokenStakingNodesManager = init.tokenStakingNodesManager;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  STRATEGY  ----------------------------------------
+    //--------------------------------------------------------------------------------------
+    
+    /**
+     * @notice Stakes specified amounts of assets into a specific node on EigenLayer.
+     * @param nodeId The ID of the node where assets will be staked.
+     * @param assets An array of ERC20 tokens to be staked.
+     * @param amounts An array of amounts corresponding to each asset to be staked.
+     */
+    function stakeAssetsToNode(
+        uint256 nodeId,
+        IERC20[] calldata assets,
+        uint256[] calldata amounts
+    ) external onlyRole(STRATEGY_CONTROLLER_ROLE) nonReentrant {
+        if (assets.length != amounts.length) {
+            revert LengthMismatch(assets.length, amounts.length);
+        }
+
+        ITokenStakingNode node = tokenStakingNodesManager.getNodeById(nodeId);
+        if (address(node) == address(0)) {
+            revert InvalidNodeId(nodeId);
+        }
+
+        IStrategy[] memory strategiesForNode = new IStrategy[](assets.length);
+        for (uint256 i = 0; i < assets.length; i++) {
```

Comment:
cache the lenght `assetsLength = assets.length`. This var can then also be reused for the loop on line 160